### PR TITLE
feat(remap): improve test harness set-up

### PR DESCRIPTION
**Pull request by JeanMertz**
As we're getting close to shipping VRL, we're seeing some edge-cases and other bugs that we haven't fixed yet. We have a relatively extensive set of unit tests, but there's many corner cases that aren't tested yet.

The problem is that you need to be somewhat familiar with our code base to be able to add new tests.

This isn't the case with the newly added test harness.

So, I figured it'd be a good step to create some more consistency in the harness, and add documentation, so that others (such as @binarylogic, but really anyone who wants to contribute) can start adding tests to validate correct behaviour, and file issues based on failing tests if needed.

### CODE REVIEW

**Review by binarylogic — APPROVED** [review]
Very cool, I like this.

File: lib/remap-tests/src/main.rs [hunk 1]
```diff
@@ -10,12 +11,15 @@ fn main() {
...
 
-    for file in tests {
-        let mut test = Test::new(file.expect("file").path());
+        let mut test = Test::new(path);
 
-        print!("{:.<60}", test.name);
+        print!("{:.<70}", test.name);
 
         if test.skip {
             println!("{}", Colour::Yellow.bold().paint("SKIPPED"));
```
Comment (FungusHumungus) [inline lib/remap-tests/src/main.rs:25]: I would argue that if we skip the test, we don't need to carry on and run the test.

**Review by FungusHumungus — APPROVED** [review]
Nice.

File: lib/remap-tests/src/main.rs (hunk 1 above)
Comment (JeanMertz) [reply lib/remap-tests/src/main.rs:25]: I'd like to. I used this to support `--verbose`, which prints the output of each run without showing a diff, regardless whether the test was skipped or not.

This helps (me) during debugging.

**Review by jszwedko — APPROVED** [review]
This is very nice. Thanks for the documentation around it too!

We can do this after, but should we add it to CI?

**Comment by JeanMertz** [comment]
> We can do this after, but should we add it to CI?

Yeah, I was planning on doing one more PR in which I add a bunch of tests I have lying around locally (not all that are pushed to Git currently pass), after that, I'd love to get this running in the CI 👍
